File: java/70. Climbing Stairs/Solution.java
class Solution {
    private Map<Integer,Integer> memo = new HashMap<>(Map.of(1, 1, 2, 2));
    public int climbStairs(int n) {
        if (memo.containsKey(n)) return memo.get(n);
        memo.put(n, climbStairs(n - 1) + climbStairs(n - 2));
        return memo.get(n);
    }
}

File: python/70. Climbing Stairs/solution.py
class Solution:
    memo = {1 : 1, 2 : 2}
    def climbStairs(self, n: int) -> int:
        if n in self.memo:
            return self.memo[n]
        else:
            self.memo[n] = self.climbStairs(n - 1) + self.climbStairs(n - 2)

        return self.memo[n]